From 1e129038cf63f619fd5b69f92057cb29c53a83d3 Mon Sep 17 00:00:00 2001
From: Pierre-Yves Chibon <pingou@pingoured.fr>
Date: Wed, 12 Aug 2020 21:19:22 +0200
Subject: [PATCH 4/9] Include the assignee in the list of people notified on a
 ticket/PR

We always include the assignee in the notifications (even on private
tickets), but so far we did not show them in the list of subscriber.

With this commit, this oversight is now fixed.

Fixes https://pagure.io/pagure/issue/4957

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>
---
 pagure/lib/query.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pagure/lib/query.py b/pagure/lib/query.py
index 8ee4df6c..498148df 100644
--- a/pagure/lib/query.py
+++ b/pagure/lib/query.py
@@ -4957,6 +4957,10 @@ def get_watch_list(session, obj):
     # Add the user of the project
     users.add(obj.project.user.username)
 
+    # Add the assignee if there is one
+    if obj.assignee:
+        users.add(obj.assignee.username)
+
     # Add the regular contributors
     for contributor in obj.project.users:
         users.add(contributor.username)
-- 
2.26.2

